Create Blog (#463)
tektoncd/website: content/en/blog/2022/graduation/index.md

---
title: Tekton Graduation
likTitle: Tekton Graduation
date: 2022-10-26
author: "Andrea Frittoli"
description: >
  From founding project to graduated, a brief history of Tekton
---

We're very happy to announce that Tekton as reached graduated status within the
Continuous Delivery Foundation (CDF). The CDF Technical Oversight Committee

> (TOC) conducted public voting decide on the graduation status for Tekton, and

Reviewer: ```suggestion
(TOC) conducted public voting to decide on the graduation status for Tekton, and
```

Might be good to add a link to the PR/list as well!

the result was unanimously positive. The Tekton community is very proud of the
results of the vote and will continue working to make Tekton better and safer
for its users.

In this blog post I will explore a bit of the history of Tekton, its "road to
graduation" and what this milestone means for the project.

## Early Days

The Tekton project has its roots in [Knative][knative], where it is initially
called "Knative Build" and later "Knative Pipeline". The project is spun off in
August 2018, when it got it's current name and a new home on GitHub as

> ["tektoncd/pipeline"][pipeline-repo]. Here's one of the project very first

Reviewer: ```suggestion
["tektoncd/pipeline"][pipeline-repo]. Here's one of the project's very first
```

commits:

```git
commit 49d2316d71e8c315e0a8fd76008bc2920f56b3c3
Author: Christie Wilson <bobcatfish@gmail.com>
Date:   Fri Aug 31 17:15:35 2018 -0700

Add pipeline strawman example

@dlorenc @ImJasonH @tejal29 @aaron-prindle and I have been working on a
strawman proposal for adding a Pipeline CRD and also for possibly
envolving the Build CRD into a slightly more generic Task CRD.

This PR demonstrates some paper prototype examples of what it could look
like to define pipelines using the CRDs described in the README.
```

A few months later, in March 2019, [Tekton is donated][cdf-creation-blog] to the
newly formed [Continuos Delivery Foundation (CDF)][cdf]. 

## Growing the community

Since then the project has thrived thanks to a rich community of contributors.
One of earlier repositories to be created is the community one, where the
project documented its governance, code of conduct and contributing guidelines.

At the end of 2019 Tekton the number of repositories had grown to eleven. A

> regular monthly release cadence is established - Pipeline reaches 9 releases; in

Reviewer: ```suggestion
regular monthly release cadence is established - Pipeline had 9 releases and in
```

Reviewer 2: I used "Pipeline has 9 releases" because everything else I kept in the present form.

March 2020 the beta version of the Tekton Pipeline API was

> [released][pipeline-beta-release]. Release automation is implemented using

Reviewer: ```suggestion
[released][pipeline-beta-release]. During this time, we also implemented the project's release automation using
```

Tekton itself.

> The "results" and "chains" project were started, and in August the Tekton Hub

Reviewer: Is there a date we can add i.e. The results and chains projects were started in 2020 or something like that

was [officially launched][tekton-hub-launch].

> The Tekton community kept growing, thanks for Tekton adopters and end-users

Reviewer: ```suggestion
The Tekton community kept growing thanks to Tekton adopters and end-users
```

contributing features, ideas and issues.

{{< imgproc contributions Fit 700x500 >}} Contributions and contributors to
Tekton Projects over time. Source <a

> href="https://tekton.devstats.cd.foundation/d/74/contributions-chart?orgId=1&from=1540938487786&to=1666996087786&var-period=m&var-metric=contributions&var-repogroup_name=All&var-country_name=All&var-company_name=All&var-company=all&viewPanel=5&theme=light&kiosk=">devstat</a>.

Reviewer: ```suggestion
href="https://tekton.devstats.cd.foundation/d/74/contributions-chart?orgId=1&from=1540938487786&to=1666996087786&var-period=m&var-metric=contributions&var-repogroup_name=All&var-country_name=All&var-company_name=All&var-company=all&viewPanel=5&theme=light&kiosk=">devstats</a>.
```

---
title: Tekton Graduation
likTitle: Tekton Graduation
date: 2022-10-26
author: "Andrea Frittoli"
description: >
  From founding project to graduated, a brief history of Tekton
---

> We're very happy to announce that Tekton as reached graduated status within the

Reviewer: ```suggestion
We're very happy to announce that Tekton has reached graduated status within the
```

Reviewer 2: Nice catch, thank you

Continuous Delivery Foundation (CDF). The CDF Technical Oversight Committee
(TOC) conducted public [voting][voting-email] to decide on the graduation status
for Tekton, and the result was unanimously positive. The Tekton community is
very proud of the results of the vote and will continue working to make Tekton
better and safer for its users.

In this blog post I will explore a bit of the history of Tekton, its "road to
graduation" and what this milestone means for the project.

## Early Days

> The Tekton project has its roots in [Knative][knative], where it is initially

Reviewer: ```suggestion
The Tekton project has its roots in [Knative][knative], where it was initially
```

> called "Knative Build" and later "Knative Pipeline". The project is spun off in

Reviewer: ```suggestion
called "Knative Build" and later "Knative Pipeline". The project was spun off in
```

August 2018, when it got it's current name and a new home on GitHub as
["tektoncd/pipeline"][pipeline-repo]. Here's one of the project's very first
commits:

```git
commit 49d2316d71e8c315e0a8fd76008bc2920f56b3c3
Author: Christie Wilson <bobcatfish@gmail.com>
Date:   Fri Aug 31 17:15:35 2018 -0700

Add pipeline strawman example

@dlorenc @ImJasonH @tejal29 @aaron-prindle and I have been working on a
strawman proposal for adding a Pipeline CRD and also for possibly
envolving the Build CRD into a slightly more generic Task CRD.

This PR demonstrates some paper prototype examples of what it could look
like to define pipelines using the CRDs described in the README.
```

A few months later, in March 2019, [Tekton is donated][cdf-creation-blog] to the
newly formed [Continuos Delivery Foundation (CDF)][cdf]. 

## Growing the community

Since then the project has thrived thanks to a rich community of contributors.

> One of earlier repositories to be created is the community one, where the

Reviewer: Link to the community repo maybe?

project documented its governance, code of conduct and contributing guidelines.

At the end of 2019 Tekton the number of repositories had grown to eleven. A

> regular monthly release cadence is established - Pipeline has 9 releases and in

Reviewer: ```suggestion
regular monthly release cadence was established - Pipeline has 9 releases and in
```

> regular monthly release cadence is established - Pipeline has 9 releases and in

Reviewer: ```suggestion
regular monthly release cadence is established - Pipeline had 9 releases and in
```

the Tekton Hub was [officially launched][tekton-hub-launch].

The Tekton community kept growing thanks to Tekton adopters and end-users
contributing features, ideas and issues.

{{< imgproc contributions Fit 700x500 >}} Contributions and contributors to
Tekton Projects over time. Source <a
href="https://tekton.devstats.cd.foundation/d/74/contributions-chart?orgId=1&from=1540938487786&to=1666996087786&var-period=m&var-metric=contributions&var-repogroup_name=All&var-country_name=All&var-company_name=All&var-company=all&viewPanel=5&theme=light&kiosk=">devstats</a>.
{{< /imgproc >}}

## Focus on security

One year later, it's now 2021, Tekton has matured considerably, while remaining
true to its nature of having a small footprint and giving users full flexibility
in how they setup their CI/CD system through Tekton.

This very flexibility has enabled Tekton to become the base for the
implementation of more opinionated services on top, ranging from open source
projects, and cloud services as well as end-user platforms for DevOps services.

It's now prime time to focus more thoroughly on security! In July 2021 the

> Tekton Vulnerability Team is formed. Tekton pipelines release v0.29 is the first

Reviewer: ```suggestion
Tekton Vulnerability Team was formed. Tekton pipelines release v0.29 is the first
```

> Tekton Vulnerability Team is formed. Tekton pipelines release v0.29 is the first

Reviewer: ```suggestion
Tekton Vulnerability Team is formed. Tekton pipelines release v0.29 was the first
```
